src/Worker.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+/// <summary>
+/// Request used to make an HTTP call through Durable Functions.
+/// </summary>
+public class DurableHttpRequest
```

Comment:
Leaving a note to add support for the following properties as well:

- ITokenSource tokenSource
- bool asynchronousPatternEnabled
- TimeSpan? timeout
- HttpRetryOptions httpRetryOptions

Reply:
`ITokenSource` is the one I don't know if we will be able to support fully - or if we should. It is a lot more complex because these are two different assemblies and processes, so the types won't line up. We can at least add support for managed identities. But not custom `ITokenSource`.

Reply:
I see, so you're saying to only add a `ManagedIdentityTokenSource` for .NET Isolated? This would mean that the `CallHttpAsync` API would have different arguments for .NET In proc and Isolated though. Is that not confusing for customers?

In proc:
`CallHttpAsync(.. ITokenSource tokenSource, ..)`

Isolated:
`CallHttpAsync(.. ManagedIdentityTokenSource tokenSource ..)`

Reply:
The arguments would need to be different yes, but not exactly how you mentioned. We wouldn't be exposing the direct types in the out of proc API as it is a different process. We would have to design and implement some semantic way to indicate what auth behavior they want to happen in the host process.

Reply:
Ok, I think that's out of scope for this PR then, but we can add it in the future.